### PR TITLE
[optimizer] Add optimizer wrapped 

### PR DESCRIPTION
**Pull request by kparichay**
- Add optimizer wrapped which wraps the opitmizer and the learning rate
scheduler.
In order to be backward compatible, each optimizer must support setting
the learning rate, decay rate and decay steps, even for new optimizers.
To make this extensible without each optimizer storing this information
and merging with the learning rate schedulers, and not creating new
interfaces, optimizer wrapped is added.
Optimizer wraps around optimizer, and owns both the optimizer and
learning rate scheduler. If the properties of LR or decay are passed to
the optimizer, they are intercepted by the optimizer wrapped and passed
to the learning rate scheduler appropriately.
- Update to use the optimizer wrapped instead of optimizer directly, and
prepare for the using learning rate scheduler.
- Update the optimizer to use the learning rate scheduler in the training
and weight updation.
- cleanup the optimizers.

See Also #1776

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: nntrainer/optimizers/optimizer_wrapped.cpp [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  auto remain_props = loadProperties(values, props);
+  optimizer->setProperty(remain_props);
+}
+
+double OptimizerWrapped::getLearningRate(size_t iteration) {
+  return lr_sched->getLearningRate(iteration);
+}
+
+void OptimizerWrapped::applyGradient(RunOptimizerContext &context) {
+  // optimizer->applyGradient(context);
```
Comment (zhoonit) [inline nntrainer/optimizers/optimizer_wrapped.cpp]: comment intended?

File: nntrainer/optimizers/optimizer_wrapped.cpp [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+}
+
+void OptimizerWrapped::exportTo(Exporter &exporter,
+                                const ExportMethods &method) const {
+  // optimizer->exportTo(exporter, method);
```
Comment (zhoonit) [inline nntrainer/optimizers/optimizer_wrapped.cpp]: comment intended?

File: nntrainer/optimizers/optimizer_wrapped.cpp (hunk 1 above)
Comment (kparichay) [reply nntrainer/optimizers/optimizer_wrapped.cpp]: yes, this will be enabled once the optimizer is set in upcoming PR.

File: nntrainer/optimizers/optimizer_wrapped.cpp (hunk 2 above)
Comment (kparichay) [reply nntrainer/optimizers/optimizer_wrapped.cpp]: yes, this will be enabled once the optimizer is set in upcoming PR.

**Review by zhoonit — APPROVED** [review]
LGTM, could you share briefly

**Review by lhs8928 — APPROVED** [review]
LGTM

**Comment by jijoongmoon** [comment]
This pr is included in #1871
